services/ui-src/src/components/modals/DeleteEntityModal.tsx

```diff
@@ -70,18 +70,20 @@ export const DeleteEntityModal = ({
     const entityTypes: string[] =
       typeof entityType === "string" ? [entityType] : (entityType as string[]);
 
+    const entityName = entityTypes[0];
     const updatedEntities = removeSelectedEntity(
       structuredClone(report?.fieldData),
       entityTypes[entityTypes.length - 1],
       selectedEntity?.id
     );
 
+    const tempEntities = { ...updatedEntities };
     await updateReport(reportKeys, {
       metadata: {
         lastAlteredBy: full_name,
         status: ReportStatus.IN_PROGRESS,
       },
-      fieldData: { ...updatedEntities },
+      fieldData: { [entityName]: tempEntities[entityName] },
```

Comment:
nit: not sure if its necessary to store the updatedEntities in temp variable when its only for data retrieval and theres no modification after. you can just use it as:

`fieldData: { [entityName]: updatedEntities [entityName] },`

Reply:
you were right. updated.